### PR TITLE
Initial PR

### PR DESCRIPTION
**Pull request by mikonc**

### CODE REVIEW

File: user_management/datadog_users/manage_datadog_users.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+from datadog_api_client.v2.model.users_type import UsersType
+from .admin_users import ADMIN_USERS
+
+
+class ManageDatadogUsers:
+    def __init__(self, api_key, app_key, datadog_url):
+        os.environ["DD_SITE"] = datadog_url
+        os.environ["DD_API_KEY"] = api_key
+        os.environ["DD_APP_KEY"] = app_key
+        self.api_client = ApiClient(Configuration())
```
Comment (grzegorzpro) [inline user_management/datadog_users/manage_datadog_users.py:16]: Judging from the docs (https://datadoghq.dev/datadog-api-client-python/datadog_api_client.html#module-datadog_api_client.configuration) it should be possible to provide the credentials and url using the `Configuration` object instead of setting global env variables

File: user_management/datadog_users/manage_datadog_users.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+        return role_data["attributes"].name
+
+    def disable_multiple_users(self, user_list):
+        disabled_users = []
+        for user in user_list:
+            self.disable_user_account(user["id"])
+            disabled_users.append(user)
+
+        return disabled_users
```
Comment (grzegorzpro) [inline user_management/datadog_users/manage_datadog_users.py:111]: I think you are creating and returning exactly the same list as `user_list`, why is this needed?

File: user_management/datadog_users/manage_datadog_users.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+    def remove_user_from_role(self, user_id, role_id):
+        body = RelationshipToUser(
+            data=RelationshipToUserData(id=user_id, type=UsersType("users"))
+        )
+
+        api_instance = RolesApi(self.api_client)
+        api_instance.remove_user_from_role(role_id=role_id, body=body)
+
+    def list_role_users(self):
```
Comment (grzegorzpro) [inline user_management/datadog_users/manage_datadog_users.py:178]: I think this method is unused, do you want to keep it for some reason?

File: user_management/datadog_users/manage_datadog_users.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+                )
+                self.remove_user_from_role(
+                    user["id"], self.user_roles["Datadog Admin Role"]
+                )
+                self.remove_user_from_role(
+                    user["id"], self.user_roles["Datadog Standard Role"]
+                )
+                user["role"] = "Datadog Read Only Role"
+
+            downgraded_users.append(user)
```
Comment (grzegorzpro) [inline user_management/datadog_users/manage_datadog_users.py:131]: Same question as above - this list will be exactly the same as `user_list`, or was this line supposed to be inside the `if`?

File: user_management/datadog_users/manage_datadog_users.py [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+        for user in users.data:
+            user_email = user["attributes"].email
+            user_status = user["attributes"].status
+            user_role_id = user["relationships"].roles.data[0].id
+
+            if user_role_id in org_user_roles.values():
+                user_role_name = list(org_user_roles.keys())[
+                    list(org_user_roles.values()).index(user_role_id)
+                ]
+            else:
```
Comment (grzegorzpro) [inline user_management/datadog_users/manage_datadog_users.py:35]: Do we know why the role might sometimes be missing from `org_user_roles`? Isn't it because the `get_organization_roles` method below doesn't use pagination?

File: user_management/datadog_users/manage_datadog_users.py [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+                        "status": user_status,
+                        "role": user_role_name,
+                        "nordcloud user": is_nordcloud_user,
+                        "id": user.id,
+                    }
+                )
+
+        return self.user_list, self.users_to_downgrade, self.users_to_disable
+
+    def get_organization_roles(self):
```
Comment (grzegorzpro) [inline user_management/datadog_users/manage_datadog_users.py:88]: You could change this into a cached property to avoid passing around the returned value. You'd need to decorate this method with `@cached_property`, rename it to `organization_roles` and then `get_organization_users` could just do `self.organization_roles` instead of requiring the roles as an argument :) It's a bit of an advanced trick, but very useful sometimes

File: user_management/datadog_users/manage_datadog_users.py (hunk 5 above)
Comment (mikonc) [reply user_management/datadog_users/manage_datadog_users.py:35]: It might be missing in a case where a customer is using a custom roles - Read only, standard and Admin are there by default.

**Review by grzegorzpro — APPROVED** [review]
